doc/developer/style-informal.md

```diff
@@ -0,0 +1,66 @@
+about how it will behave when the data size is 10x, 100x, or even 1000x larger.
+What are the bottlenecks? Will it still be performant? Will it still be maintainable?
+
+This doesn't mean that the code you write today has to be perfect for all future scales, but
+it should be easy to change and adapt as the scale changes. Documentation should point out
+any assumptions or limitations that the code has regarding scale. Leave breadcrumbs for your
+future self and others who will work on the code later.
+
+A common way to think about scalability is to keep time and space complexity in mind.
+For example, if you are writing a function that takes 50 microseconds per item, it is
+fine to write a O(n^2) algorithm for very small n, but it will quickly become a problem.
+The following table demonstrates how different time complexities scale with increasing input sizes:
+
+| Input Size (n) | O(1)     | O(log n) | O(n)  | O(n log n) | O(n^2)     |
+|----------------|----------|----------|-------|------------|------------|
+| 10             | constant | 166us    | 500us | 1.66ms     | 5ms        |
+| 100            | constant | 333us    | 5ms   | 33ms       | 500ms      |
+| 1,000          | constant | 500us    | 50ms  | ~500ms     | 50s        |
+| 10,000         | constant | 665us    | 500ms | ~6.7s      | 84 minutes |
+| 100,000        | constant | 831us    | 5s    | ~83s       | 139h       |
+| 1,000,000      | constant | 1ms      | 50s   | ~1000s     | 1.6 years  |
+
+## Prefer simplicity to complexity
+
+When writing code, you should always prefer simplicity to complexity. Rely on Rust's type system
+to enforce invariants, rather than writing complex logic to check for them at runtime.
+
+* If a function gets too hard to explain, consider breaking it up into smaller functions.
+* If a change requires extensive documentation to explain, consider simplifying the change.
+* Consider separating a function into two functions if it has multiple responsibilities.
+  An indicator could be a boolen argument and if-else branches that depend on it.
+
+## Listen for feedback
+
+Language is ambiguous, and different people have different interpretations of the same words.
+When you receive feedback on your change, try to understand the perspective of the person giving the feedback.
+Ask questions if you don't understand their point of view.
+It's better to ask questions and clarify misunderstandings than to assume you know what they mean.
+
+## Solving problems over quick fixes
+
+We sometimes need to ship fixes quickly, but we should always try to solve the underlying problem.
+Staggering a fix into multiple changes is OK, as long as we don't forget halfway through.
+Accumulating technical debt is easy, but it makes future changes harder and more error-prone.
+
+## Revert rather than fix-forward
+
+If release contains a bug, prefer reverting the offending change rather than trying to fix it in place.
```

Comment:
hm I think this point might require a bit more nuance. I can think of situations where both fixing forward and reverting can be preferable to one another